WordPress/src/test/java/org/wordpress/android/util/SnackbarSequencerConcurrentTest.kt

```diff
@@ -76,42 +81,91 @@ class SnackbarSequencerConcurrentTest {
         // Then
         verify(wpSnackbar, times(1)).show()
         advanceTimeBy(checkPoints[0])
-        verify(wpSnackbar, times(1)).show()
+        verify(wpSnackbar, times(2)).show()
         advanceTimeBy(checkPoints[1])
+        verify(wpSnackbar, times(3)).show()
+        advanceTimeBy(checkPoints[2])
+
+        // Nothing else shown
+        verify(wpSnackbar, times(3)).show()
+        verifyNoMoreInteractions(wpSnackbar)
+    }
+
+    @Test
+    fun `snackbars are not shown until previous duration elapsed`() = runBlockingTest(coroutineContext) {
+        // Given
+        val items = getItems()
+        val checkPoints = getCheckPoints(items)
+
+        // When
+        for (item in items) {
+            sequencer.enqueue(item)
+        }
+
+        // Then
+        verify(wpSnackbar, times(1)).show()
+        advanceTimeBy(checkPoints[0] - 2 * SNACKBAR_DURATION_MARGIN)
+        verify(wpSnackbar, times(1)).show()
+        advanceTimeBy(2 * SNACKBAR_DURATION_MARGIN)
         verify(wpSnackbar, times(2)).show()
+        advanceTimeBy(checkPoints[1])
+        verify(wpSnackbar, times(3)).show()
         advanceTimeBy(checkPoints[2])
+
+        // Nothing else shown
         verify(wpSnackbar, times(3)).show()
+        verifyNoMoreInteractions(wpSnackbar)
+    }
+
+    @Test
+    fun `snackbars beyond capacity are not shown`() = runBlockingTest(coroutineContext) {
+        // Given
+        val items = mutableListOf<SnackbarItem>()
+
+        items.addAll(getItems())
+        items.addAll(getItems())
+        items.addAll(getItems())
+
+        // When
+        for (item in items) {
+            sequencer.enqueue(item)
+        }
+
+        // Then
+        advanceTimeBy((max(TEST_DURATIONS) * items.size).toLong())
+        verify(wpSnackbar, times(QUEUE_SIZE_LIMIT + 1)).show()
+        verifyNoMoreInteractions(wpSnackbar)
     }
 
-    private fun getCheckPoints(items: List<SnackbarSequencerInfo>): List<Long> {
+    private fun getCheckPoints(items: List<SnackbarItem>): List<Long> {
         return listOf(
-            getSnackbarDurationMs(items[0]),
-            DURATION_EXTRA_MARGIN + getSnackbarDurationMs(items[1]),
-            DURATION_EXTRA_MARGIN + getSnackbarDurationMs(items[2])
+            SNACKBAR_DURATION_MARGIN + getSnackbarDurationMs(items[0]),
```

Comment:
Could you please elaborate what is the purpose of `SNACKBAR_DURATION_MARGIN` here?

Reply:
The intent was to have the check points not fall exactly on the snachbar durations but just a bit later so to not making the checks exactly on the edge cases. Let me know if it makes any sense 😊.

Reply:
I see. Imo it makes the tests more complicated and they'll work even when you remove it. But I'm find with keeping it. Thanks for the explanation.

Reply:
After the discussion we had on slack. I think we might want to consider do the following
- remove SNACKBAR_DURATION_MARGIN
- rename "getCheckpoints" to "getDurations"
- replace advanceTimeBy(checkPoints[0]) with advanceTimeBy(durations[0] + 1)

Reply:
see [here](https://github.com/wordpress-mobile/WordPress-Android/pull/10856#discussion_r351933046)